servicetest/coredump/test_coredump.py

```diff
@@ -0,0 +1,175 @@
+        that should be used with this test module. The agent fixture expects a list
+        of StandardManifest and/or DefaultManifest objects.
+    """
+    return [default_manifest]
+
+@pytest.fixture
+def create_testoutput_dir(scope="module"):
+    """ Create a directory for the generated test files.
+
+        This directory is ignored by git but it's nice to have
+        somewhere locally to store test output to support
+        troubleshooting etc.
+    """
+    if not os.path.exists(TESTOUTPUT_DIR):
+        os.makedirs(TESTOUTPUT_DIR)
+
+@pytest.fixture
+def create_coredump_dir(scope="module"):
+    """ Create the directories needed for getting the core dumps
+    """
+
+    # Create the core dump directory if it doesn't already exist
+    if not os.path.exists(coredump_path()):
+        os.makedirs(coredump_path())
+
+    yield
+
+    shutil.rmtree(coredump_path())
+    assert(not os.path.exists(coredump_path()))
+
+@pytest.fixture
+def set_core_pattern(scope="function"):
+    """ Check the core pattern and set it to something usable for this test
+
+        Also reset the pattern at test end
+    """
+
+    pattern_path = "/proc/sys/kernel/core_pattern"
+    test_pattern = coredump_path() + "/coredump"
+    old_pattern = ""
+
+    with open(pattern_path, "r") as core_file:
+        old_pattern = core_file.read().strip()
+
+    with open(pattern_path, "w") as core_file:
+        core_file.write(test_pattern)
+
+    with open(pattern_path, "r") as core_file:
+        assert(core_file.read().strip() == test_pattern)
+
+    yield test_pattern
+
+    # Teardown
+    with open(pattern_path, "w") as core_file:
+        core_file.write(old_pattern)
+
+    with open(pattern_path, "r") as core_file:
+        assert(core_file.read().strip() == old_pattern)
+
+@pytest.mark.usefixtures("testhelper", "dbus_launch", "create_testoutput_dir",
+                         "create_coredump_dir", "agent", "assert_no_proxy")
+class TestCore(object):
+    """ This suite tests that core dumps are created properly.
+
+        The idea is to have a default capability that mounts whatever directory one
+        wants to have available for core dumps into the container, and to make sure
+        they end up there.
+    """
+
+    def test_core_is_dumped(self, set_core_pattern):
+        """ Test that the core is dumped
+        """
+        sc = Container()
+
+        try:
+            sc.start(DATA)
+
+            # Start without setting any capabilities
+            pid = sc.launch_command("python " +
+                              sc.get_bind_dir() +
+                              "/testhelper.py --test-dir " +
+                              sc.get_bind_dir() + "/testoutput" +
+                              " --wait-for-core-dump " + coredump_path()
+                              )
+            # Let the helper run
+            time.sleep(0.25)
+
+            # Generate a segfault, causing core to be dumped
+            os.kill(pid, signal.SIGSEGV)
```

Comment:
But isn't it nice to have the test code explicit in what the sequence look like? If the core dump is generated implicitly by using a certain helper it can perhaps be confusing for the reader.